doc/how-to/vshard_quick.rst

```diff
@@ -1,160 +1,525 @@
+        :language: lua
+        :start-at: function get_band
+        :dedent:
+
+The resulting ``storage.lua`` file should look as follows:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/storage.lua
+    :language: lua
+    :dedent:
+
+
+..  _vshard-quick-start-router-code:
+
+Adding router code
+~~~~~~~~~~~~~~~~~~
+
+1.  Open the ``router.lua`` file and load the ``vshard`` module as follows:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/router.lua
+        :language: lua
+        :start-at: local vshard
+        :end-at: local vshard
+        :dedent:
+
+2.  Define the ``put`` function used to write data to a storage:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/router.lua
+        :language: lua
+        :start-at: function put
+        :end-before: function get
+        :dedent:
+
+    The following ``vshard`` router functions are used:
+
+    *   :ref:`vshard.router.bucket_id_mpcrc32() <router_api-bucket_id_mpcrc32>`: Calculates a bucket ID value using a hash function.
+    *   :ref:`vshard.router.callrw() <router_api-callrw>`: Inserts a tuple to a storage identified the generated bucket ID.
+
+3.  Create the ``get`` function for getting data:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/router.lua
+        :language: lua
+        :start-at: function get
+        :end-before: function insert_data
+        :dedent:
+
+    Inside this function, :ref:`vshard.router.callro() <router_api-callro>` is called to get data from a storage identified the generated bucket ID.
+
+4.  Finally, create the ``insert_data()`` function that inserts sample data into the created space:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/router.lua
+        :language: lua
+        :start-at: function insert_data
+        :dedent:
+
+The resulting ``router.lua`` file should look as follows:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/router.lua
+    :language: lua
+    :dedent:
+
+
+
+..  _vshard-quick-start-build-settings:
+
+Configuring build settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Open the ``sharded_cluster-scm-1.rockspec`` file and add the following content:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/sharded_cluster-scm-1.rockspec
+    :language: none
+    :dedent:
+
+The ``dependencies`` section includes the specified version of the ``vshard`` module.
+To install dependencies, you need to :ref:`build the application <vshard-quick-start-building-app>`.
+
+
+.. _vshard-quick-start-building-app:
+
+Building the application
+------------------------
+
+In the terminal, open a directory where the :ref:`tt environment is created <vshard-quick-start-creating-app>`.
```

Comment:
`directory where the tt environment is created` > `the tt environment directory`?